azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/http/AzureBoardsOAuthService.java

```diff
@@ -0,0 +1,86 @@
+package com.synopsys.integration.azure.boards.common.http;
+
+import java.io.IOException;
+import java.net.Proxy;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.AuthorizationCodeTokenRequest;
+import com.google.api.client.auth.oauth2.BearerToken;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.auth.oauth2.StoredCredential;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.Base64;
+import com.google.api.client.util.store.DataStoreFactory;
+import com.synopsys.integration.exception.IntegrationException;
+
+public class AzureBoardsOAuthService {
```

Comment:
The API for this class will definitely change, but for now, I'm going to leave this as a POC.